examples/generalized_linear_models/GLM-robust.ipynb

```diff
@@ -16,7 +16,7 @@
     "\n",
```

Comment:
<p>wanted to convert</p><pre class="ql-syntax" spellcheck="false">pm.plot_posterior_predictive_glm(trace, samples=100, label="posterior predictive regression lines")
</pre><p>to use ArviZ and get rid of the DeprecationWarning. I figured I could use <code>az.plot_ppc(trace);</code> after calling <code>model.predict()</code>, but that ends up creating two different figures/plots, so how do I work around that?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/205/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
<p>First some context: this plot is the posterior distribution of the regression lines. What do these regression lines represent? The mean value of the conditional distribution of Y given a certain value of X. So what we actually have above is not  the posterior predictive distribution, because the posterior predictive distribution is not in the scale of the mean, but on the scale of the observed variable Y (I come back to this in the next comment).</p><p> </p><p>So, what we need to get this posterior distribution of the mean in Bambi is the following</p><p> </p><p> </p><pre class="ql-syntax" spellcheck="false"># Simply a grid going from 0 to 1
# This isn't strictly necessary, but will be helpful
# since values of x are already sorted
new_data = pd.DataFrame({"x": np.linspace(0, 1, 100)})


# Computes the posterior regression lines, and stores the values in posterior["y_mean"]
model.predict(idata, data=new_data)


# Stack chains (2 in my computer)
posterior_y_mean = idata.posterior["y_mean"].values.reshape((2 * 2000, 100))
</pre><p> </p><pre class="ql-syntax" spellcheck="false"># Create the plot
plt.figure(figsize=(7, 5))

# The [:, :100] selects the 100 samples in your original plot
plt.plot(new_data["x"], posterior_y_mean.T[:, :100], color="k", alpha=0.3, lw=0.5)

plt.plot(x_out, y_out, "x", label="data")

plt.plot(x, true_regression_line, label="true regression line", lw=3.0, c="y")

plt.legend(loc=0);
</pre><p> </p><p>Unfortunately it is not as easy as using <code>pm.plot_posterior_predictive_glm</code>  yet... Hopefully, this kind of plot are going to be built-in Bambi in the future.</p>

Reply:
<p>On the other hand, if you want the posterior predictive distribution, you do</p><p> </p><p><code>model.predict(idata, kind="pps")</code></p><p><code>az.plot_ppc(idata)</code></p><p> </p><p><span class="ql-cursor">﻿</span>This overlays the observed distribution of the data with the posterior predictive distribution. Ideally, the posterior predictive samples should not deviate much from the shape of the observed data.</p>

Reply:
<p>thanks a lot for the explanation, will use this code.</p><p><code>model.predict(idata, kind="pps")</code></p><p><code>az.plot_ppc(idata)</code></p><p>is what I was trying to do without realizing that we are not trying to plot the posterior predictive distribution.</p><p> </p>

Reply:
<p>This same kind of spaghetti plot can already be generated with ArviZ, see https://arviz-devs.github.io/arviz/api/generated/arviz.plot_lm.html and https://utkarsh-maheshwari.medium.com/how-to-use-plot-lm-part-i-1f1cb099503e. The function is already available in ArviZ dev version which should be released soon</p>